java-api/src/main/kotlin/io/spine/validation/api/MessageValidator.kt

```diff
@@ -0,0 +1,141 @@
+ * definitions and the compiled class files (.class) for these messages.
+ * As these classes are pre-generated, consumers cannot modify their underlying
+ * `.proto` files to define validation constraints and the Validation library
+ * cannot use code generation to enforce the constraints.
+ *
+ * Thus, the library effectively deals with the two types of messages:
+ *
+ * 1. **Local messages** are message types for which end-users generate and control
+ *    the Java/Kotlin classes by compiling the corresponding `.proto` definitions
+ *    within their own codebase. For such messages, the Validation library allows
+ *    declaring validation constraints and enforces them with the generated code.
+ *
+ * 2. **External messages** are message types for which end-users do not generate or control
+ *    the Java/Kotlin classes. Because the classes are already generated, users cannot modify
+ *    the underlying `.proto` definitions and add validation options at compile time.
+ *
+ * ## Validation of external messages
+ *
+ * The Validation library provides a mechanism that allows validating of
+ * the external messages, **which are used for fields within local messages**.
+ * Implement this interface and annotate the implementing class with
+ * the [@Validator][Validator] annotation, specifying the message type to validate.
+ *
+ * For each field of type [M] within any local message, the library will invoke
+ * the [MessageValidator.validate] method when validating the local message.
+ *
+ * The following Protobuf field types are supported:
+ *
+ * 1. Singular fields of type [M].
+ * 2. Repeated field of type [M].
+ * 3. Map field with values of type [M].
+ *
+ * An example of the validator declaration for the `Earphones` message:
+ *
+ * ```kotlin
+ * @Validator(Earphones::class)
+ * public class EarphonesValidator : MessageValidator<Earphones> {
+ *     public override fun validate(message: Earphones): List<DetectedViolation> {
+ *         return emptyList() // Always valid.
+ *     }
+ * }
+ * ```
+ *
+ * Please note that standalone instances of [M] and fields of [M] type that occur in
+ * other external messages **will not be validated**.
+ *
+ * Consider the following example:
+ *
+ * ```proto
+ * // Brings the `Earphones` message from dependencies.
+ * // Suppose we don't control the generated code of declarations from this file.
+ * import "earphones.proto";
+ *
+ * // A locally-declared message.
+ * message WorkingSetup {
+ *
+ *     // The field of external message type.
+ *     Earphones earphones = 1;
+ * }
+ * ```
+ *
+ * Supposing that the Validation library applied to the module where both `WorkingSetup`
+ * and `EarphonesValidator` classes are declared, then the generated code of `WorkingSetup`
+ * will apple the validator to each instance passed to the `WorkingSetup.earphones` field.
+ *
+ * ## Implementation
+ *
+ * The message validator does not have restrictions upon how exactly the message
```

Comment:
Let's start with a positive sentence, not with the one built around the negation.